merkle/map_hasher.go

```diff
@@ -0,0 +1,41 @@
+package merkle
+
+import (
+	"github.com/google/trillian"
+)
+
+// MapHasher is a specialised TreeHasher which also knows about the set of
+// "null" hashes for the unused sections of a SparseMerkleTree.
+type MapHasher struct {
```

Comment:
Possibly not; I'm not sure whether/why that's there, but I'll leave a TODO - it might be we need a configurable `TreeHasher` for these differences.